service/src/main/java/org/jboss/aerogear/unifiedpush/service/util/GCMTopicManager.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.service.util;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import org.jboss.aerogear.unifiedpush.api.AndroidVariant;
+import org.jboss.aerogear.unifiedpush.api.Category;
+import org.jboss.aerogear.unifiedpush.api.Installation;
+import org.jboss.aerogear.unifiedpush.utils.AeroGearLogger;
+
+/**
+ *  Utility used for Android installations which need to be unsubscribed from topics
+ */
+public class GCMTopicManager {
+
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(GCMTopicManager.class);
+
+    // Instance ID API URL
+    public static final String IIDURL = "https://iid.googleapis.com/iid/v1/";
+
+    private AndroidVariant variant;
+
+    public GCMTopicManager(AndroidVariant variant) {
+        this.variant = variant;
+    }
+
+    /**
+     * Unsubscribes device from single category(topic)
+     *
+     * @param installation Installation object containing correct variant property of AndroidVariant class
+     * @param categoryToUnsubscribe category(topic) that device should be unsubscribed from
+     */
+    public void unsubscribe(Installation installation, Category categoryToUnsubscribe) {
+
+        String url = "";
+        try {
+            url = IIDURL + installation.getDeviceToken() + "/rel/topics/" + URLEncoder.encode(categoryToUnsubscribe.getName(), "UTF-8");
+        } catch (UnsupportedEncodingException e1) {
+            //
+        }
+
+        try {
+            delete(url);
+        } catch (IOException e) {
+            logger.fine("Unregistering device from topic was unsuccessfull");
+        }
+    }
+
+    /**
+     * Sends DELETE HTTP request to provided URL. Request is authorized using Google API key.
+     *
+     * @param urlS target URL string
+     */
+    protected int delete(String urlS) throws IOException {
```

Comment:
> 1) schedule a retry or 2) make something pop up in the UI or 3) ??? (open for suggestions)

I think, we should not retry, just impriove the server side logging, in failure cases

Doing a pop up on the UI ... also not sure about it, because the update of the installations (re-registering) is triggering this, nothing really directly from the console.

I see the need for a better visiulation of general errors and better access to logs in future versions.

Do we have a ticket to improve our (real time) logs story ?